### PR TITLE
WTB-422 - Adding validation

### PR DESCRIPTION
**Pull request by cemusta**
- added input validation for patch requests.
- moved controllers into api folder

### CODE REVIEW

File: src/api/controller/AdmissionConfigController.ts [hunk 1]
```diff
@@ -41,26 +45,37 @@ export class AdmissionConfigController {
...
       }
 
-      const updateData = req.body;
-      // TODO : Do input validation here.
-      const config = await this.configService.updateConfig(id, updateData);
+      const input = plainToClass(AdmissionConfigPatch, req.body, { excludeExtraneousValues: true });
+      logger.debug({ input }, 'input');
+
+      const errors = validateSync(input, { skipMissingProperties: true });
+      if (errors.length > 0) {
```
Comment (lutzk) [inline src/api/controller/AdmissionConfigController.ts:63]: maybe return error to client?
might be better to know what exactly is wrong

Comment (cemusta) [reply src/api/controller/AdmissionConfigController.ts:63]: Good idea

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "type-check": "tsc --noEmit -p tsconfig.json"
   },
   "nodemonConfig": {
-    "exec": "nps start.dev",
+    "exec": "nps start",
```
Comment (rafaelpinabew) [inline package.json:25]: I think this should stay as nps start.dev, no?

Comment (cemusta) [reply package.json:25]: I had some weird errors with ts-node yesterday. Can undo this but need to be sure why it was acting up.

**Review by rafaelpinabew — APPROVED** [review]

**Review by darrylhebbes — APPROVED** [review]

**Pull request merged**